student-test/assignment1/Test_validate_username.c

```diff
@@ -14,9 +14,9 @@
 */
 void test_validate_my_username()
 {
-    /**
-     * TODO: Replace the line below with your code here as described above to verify your /conf/username.txt 
-     * config file and my_username() functions are setup properly
-     */
-    TEST_ASSERT_TRUE_MESSAGE(false,"AESD students, please fix me!");
+    const char *ch1;
+    const char *ch2;
+    ch1 = my_username();
+    ch2 = malloc_username_from_conf_file();
```

Comment:
@sujoyray-ucolorado what happens to the string returned by `ch2`?

Reply:
Hi Dan, Sorry, I did not understand the question. Do you mean that I should be checking NULL pointer? If that is question, yes, I agree.

Reply:
I'll discuss in class tomorrow, but take a look at the contract on `malloc_username_from_conf_file` and the responsibilities of the caller.

Reply:
Ok, I will. Thank you

Reply:
I got it now. The ch2 needs to be freed :-). By bad. I did not look into the API